satpy/readers/__init__.py

```diff
@@ -779,8 +780,11 @@ def _get_compression(file):
 
 def open_file_or_filename(unknown_file_thing):
     """Try to open the *unknown_file_thing*, otherwise return the filename."""
-    try:
-        f_obj = unknown_file_thing.open()
-    except AttributeError:
+    if isinstance(unknown_file_thing, (pathlib.WindowsPath, pathlib.PosixPath)):
```

Comment:
Are `PosixPath` and `WindowsPath` subclasses of `pathlib.Path`? Can we check for just that base class? Or does that not work for fsspec handling?

Reply:
Should work as well. See latest push.